control/controlclient/noise.go

```diff
@@ -65,8 +65,12 @@ func newNoiseClient(priKey key.MachinePrivate, serverPubKey key.MachinePublic, s
 		// If there is an explicit port specified use it.
 		host = u.Host
 	} else {
-		// Otherwise, controlhttp.Dial expects an http endpoint.
-		host = fmt.Sprintf("%v:80", u.Hostname())
+		// Otherwise, pass the usual ports to controlhttp.Dial
+		if u.Scheme == "https" {
```

Comment:
The serverURL almost always has `https` as the scheme (except in tests), as written this will make TLS the default mode.

Reply:
I might've misunderstood what this code does, but I don't see how this would make TLS the default option for Noise.

The rationale for this is that we do use https for our integration tests, but usually not in 443...

Reply:
@maisem would it be possible for you to have another look at this?

Reply:
Sorry this dropped off my radar.

The way it's written as it is, is to make it possible for tailscale to run in as many environments as possible. The way this works is by trying to reach out to the noise endpoint first over port 80 and then 500ms later over 443.
It prefers port 80 to avoid double encryption as noise is already encrypted and with TLS it is going to do noise-over-TLS, which isn't very useful except for environments where HTTP is completely blocked.

Having said that, you mentioned your requirement is to make this work in your integration test env which uses TLS. For that you could either not use TLS for the noise part, or if you really want to test it noise-over-TLS then I'd recommend adding an additional field to pass in as the TLS port instead of saying that the HTTP and HTTPS ports are the same.